### ISSUE TITLE
WARNING: intersphinx identifier 'requests_oauthlib' is not alphanumeric

### ISSUE THREAD

**Issue by ssbarnea** [sphinx-doc/sphinx#2518]
It seems that intersphinx expects only alphanumeric identifiers but that's clearly incompatible with python module naming conventions. 

For example I have the `requests-oauthlib` package that installs the `requests_oauthlib` python module and I want to be able to inter-link with it.

How am I supposed to make this work because, so far its seems that I cannot use any of these notations and I suspect that when I will try to reference the python module the link will not work.

```
intersphinx_mapping = { 'requests_oauthlib': ('https://requests-oauthlib.readthedocs.io/en/latest/', None)}
```

And inside the documentation

```
py:mod:`requests_oauthlib` is used here!
```

**Comment by lehmannro** [comment]
From your comment it seems like your report is based on suspicions instead of actually trying it out.

The key in intersphinx_mapping is not the same thing that's used in `py:mod` — it is a "unique identifier [that] can be used to prefix cross-reference targets, so that it is clear which intersphinx set the target belongs to," as explained in [the intersphinx documentation](http://www.sphinx-doc.org/en/stable/ext/intersphinx.html#confval-intersphinx_mapping) under _New format for this config value._
In your example, you'd use:

```
:py:mod:`requests_oauthlib:requests_oauthlib` is used here!
```

...or just leave the prefix away, because that module name should be fairly unique.

You are right, though, that the warning is obnoxious.  A patch should be as easy as fixing the [condition in sphinx.ext.intersphinx](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/intersphinx.py#L271).  From a [quick look](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/util/nodes.py#L34) it seems almost any character would be valid there.

**Comment by tk0miya** [comment]
I fixed the warning at 37882b9.
It seems the condition does not have special meaning as @lehmannro said.

Thank you for reporting.
